src/Common/Auth/OpenIDConnect/Entities/UserEntity.php

```diff
@@ -90,6 +92,30 @@ protected function getAccountByPassword($userrole, $username, $password, $email
                     return false;
                 }
                 $this->setIdentifier(UuidRegistry::uuidToString($uuid));
+
+                //check if MFA required
+                $mfa = new MfaUtils($id);
+                if ($mfa->isMfaRequired() && is_null($mfaToken)) {
```

Comment:
I'm thinking you could grab the token here either by adding a method to your MfaUtils() i.e isMfaToken() then return token/null, grabbing the token from the request in your utility or grab the token from request here.

Reply:
Also will make it easier to disengage this flow from password grant.